### ISSUE TITLE
FEATURE: Edit Message

### ISSUE THREAD

**Issue by Madu-de** [Madu-de/MaduChat#119]
**Is your feature request related to a problem? Please describe.**
For now the messages arent editable. On a typo or something else the user has to send a new message to correct the message before. That makes the chat very unclean.

**Describe the solution you'd like**
I would like to implement an edit message feature to allow users to edit their messages.

**Comment by Madu-de** [comment]
@Igor-Lira Nope. If you want, you can work on it. I'll send you images of my idea tomorrow afternoon (german time)^^

**Comment by Igor-Lira** [comment]
@Madu-de Getting a bit late because of my work. But some updates, I use a mat-menu that appears on message hover. So you click in the mat-manu and it appears the options, for now it's only `edit`, but in the future could be `share`, `transfer`, etc.

I think it would be nice to have a history of modifications because maybe we need this in the future. A column with type of string array. Not sure how it works in SQL tho. For now, I would get this `message.history.length `> 0 to display `modified` on the top of the message.

For documentation to PUT message/id :

201: ok: message is edited
401: unauthorized token not valid
401: unauthorized is not a chat member
401: unauthorized is not the message owner
404: bad request, chat does not exist
404: bad requeset, message does not exist
404: bad request, message is not supported (if it is empty string, but could add some more validations later)

I have not finished to development, but those are the concepts I am implementing.

**Comment by Madu-de** [comment]
@Igor-Lira 
Sounds amazing! Thank you! I am excited👀
